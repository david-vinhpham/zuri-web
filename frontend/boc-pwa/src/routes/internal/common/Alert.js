import React, { PureComponent } from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { Modal } from 'homecredit-ui';
import { func, noop, any, objectOf } from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';

import { alerts } from 'actions/internal';
import { ALERT_TEXT } from './constant';
import s from './Alert.css';

class Alert extends PureComponent {
  static propTypes = {
    clearMessage: func,
    alert: objectOf(any),
  };

  static defaultProps = {
    clearMessage: noop,
    alert: {},
  };

  onCloseModal = () => {
    const { clearMessage } = this.props;
    clearMessage();
  };

  render() {
    const { alert } = this.props;
    const { type, message } = alert;
    const isDisplay = !!type;
    const title = type ? ALERT_TEXT[type] : '';
    return (
      <div>
        <Modal
          isOpen={isDisplay}
          onClose={this.onCloseModal}
          footer={
            <div className={s.modalFooter} onClick={this.onCloseModal}>
              <h4>{ALERT_TEXT.argee}</h4>
            </div>
          }
        >
          <div className={s.modalBody}>
            <h3 className={s.modalTitle}>{title}</h3>
            <p>{message}</p>
          </div>
        </Modal>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  clearMessage() {
    dispatch(alerts.clear());
  },
});

function mapStateToProps({ internal }) {
  const { alert } = internal;
  return { alert };
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  ),
  withStyles(s),
);

export default enhance(Alert);
