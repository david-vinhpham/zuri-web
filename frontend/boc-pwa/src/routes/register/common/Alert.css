/**
 * BOC VN (http://www.bocvietnam.com/)
 *
 * Copyright Â© 2019-present BOCVN, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

.modalFooter {
  padding: var(--b-space-3xs) var(--b-space-md);
  background-color: var(--c-brand);
  color: white;
  text-align: center;
}

.modalBody {
  text-align: center;
  color: var(--c-text);
  font-size: var(--f-text);
  font-style: italic;
  padding-bottom: var(--b-space-md);
}

.modalTitle {
  color: var(--c-text);
  font-style: normal;
}
