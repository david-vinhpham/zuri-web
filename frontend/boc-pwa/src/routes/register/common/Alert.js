import React, { PureComponent } from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { Modal } from 'homecredit-ui';
import { func, noop, string } from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';

import { clearAlert } from 'actions/saRegister';
import { ALERT_TEXT } from './constant';
import s from './Alert.css';

class Alert extends PureComponent {
  static propTypes = {
    clearMessage: func,
    title: string,
    message: string,
  };

  static defaultProps = {
    clearMessage: noop,
    title: '',
    message: '',
  };

  onCloseModal = () => {
    const { clearMessage } = this.props;
    clearMessage();
  };

  render() {
    const { title, message } = this.props;
    const isDisplay = !!message;
    return (
      <div>
        <Modal
          isOpen={isDisplay}
          onClose={this.onCloseModal}
          footer={
            <div className={s.modalFooter} onClick={this.onCloseModal}>
              <h4>{ALERT_TEXT.argee}</h4>
            </div>
          }
        >
          <div className={s.modalBody}>
            {title && <h3 className={s.modalTitle}>{title}</h3>}
            <p>{message}</p>
          </div>
        </Modal>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  clearMessage() {
    dispatch(clearAlert());
  },
});

function mapStateToProps({ saRegister }) {
  const { title, message } = saRegister.alert;
  return { title, message };
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  ),
  withStyles(s),
);

export default enhance(Alert);
