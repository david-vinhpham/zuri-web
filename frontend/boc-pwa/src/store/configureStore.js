import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { persistStore, persistReducer } from 'redux-persist';

import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

import storage from 'redux-persist/lib/storage';
import { name, version } from '../../package.json';
import rootReducer from '../reducers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const middleware = [thunk.withExtraArgument(helpersConfig)];

  const persistConfig = {
    key: 'BOCVN',
    storage,
    stateReconciler: autoMergeLevel2,
    blacklist: ['clz'],
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
    const composeEnhancers = composeWithDevTools({
      // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
      name: `${name}@${version}`,
    });

    // https://redux.js.org/docs/api/applyMiddleware.html
    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // https://redux.js.org/docs/api/createStore.html
  const store = createStore(persistedReducer, enhancer);

  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // Don't forget to remove `()` if you change reducers back to normal rootReducer.
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default()),
    );
  }

  const persistor = persistStore(store);

  return { store, persistor };
}
