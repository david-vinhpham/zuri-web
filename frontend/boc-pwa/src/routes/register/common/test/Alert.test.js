import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Alert from '../Alert';

const mockStore = configureStore();
const initialState = {
  saRegister: {
    alert: {
      title: 'Lỗi xãy ra',
      message: 'Something happend!',
    },
  },
};

it('Alert snapshot', () => {
  const store = mockStore(initialState);
  const wrapper = renderer.create(
    <Provider store={store}>
      <Alert />
    </Provider>,
  );
  const snapshot = wrapper.toJSON();

  expect(snapshot).toMatchSnapshot();
});
