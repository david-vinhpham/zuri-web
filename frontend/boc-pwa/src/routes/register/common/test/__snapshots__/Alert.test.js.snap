// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshot 1`] = `
<div>
  <div
    className=""
  >
    <div
      className="Modal_modal__2AD1n"
      onClick={[Function]}
    >
      <div
        className="Modal_modalContent__LCdOb"
      >
        <div
          className="Modal_modalBody__9wxhI"
        >
          <div
            className="modalBody"
          >
            <h3
              className="modalTitle"
            >
              Lỗi xãy ra
            </h3>
            <p>
              Something happend!
            </p>
          </div>
        </div>
        <div
          className="Modal_modalFooter__3QMxu"
        >
          <div
            className="modalFooter"
            onClick={[Function]}
          >
            <h4>
              Tôi đã hiểu
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
